spec/std/http/spec_helper.cr

```diff
@@ -61,29 +61,29 @@ end
 def run_handler(handler, &)
   done = Channel(Exception?).new
 
-  begin
-    IO::Stapled.pipe do |server_io, client_io|
-      processor = HTTP::Server::RequestProcessor.new(handler)
-      f = spawn do
-        processor.process(server_io, server_io)
-      rescue exc
-        done.send exc
-      else
-        done.send nil
-      end
+  IO::Stapled.pipe do |server_io, client_io|
+    processor = HTTP::Server::RequestProcessor.new(handler)
+    f = spawn do
+      processor.process(server_io, server_io)
+    rescue exc
+      done.send exc
+    else
+      done.send nil
+    end
 
-      client = HTTP::Client.new(client_io)
+    client = HTTP::Client.new(client_io)
 
-      begin
-        wait_until_blocked f
+    begin
+      wait_until_blocked f
 
-        yield client
-      ensure
-        processor.close
+      yield client
+    ensure
+      processor.close
+      {% unless flag?(:win32) %}
         server_io.close
-        if exc = done.receive
-          raise exc
-        end
+      {% end %}
```

Comment:
It seems one of the `IO`s was trying to flush to a closed stream if this line is present. I have no idea this broke the OAuth2 specs

Reply:
Yeah, this should require some investigation. The close behaviour should not be platform-specific.

Reply:
It looks like the current fiber is unconditionally scheduled at a later time even if `WriteFile` finishes synchronously, despite `FILE_FLAG_OVERLAPPED` being used.

Reply:
Now I'm wondering: if socket operations like `WSASend` [complete immediately](https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsasend#overlapped-socket-i-o), we also might not need to unconditionally reschedule inside `#wsa_overlapped_operation` either

Reply:
Great find! That seems like the piece we've been missing 🚀